src/Strings.h

```diff
@@ -213,8 +216,8 @@ inline void ParseRegex(IteratorT const &begin,
                        RegexT const &regexToken,
                        std::function<void (IteratorT, IteratorT)> const &callback) {
 
-    std::sregex_iterator iter(begin, end, regexToken);
-    std::sregex_iterator iterEnd;
+    std::regex_iterator<IteratorT> iter(begin, end, regexToken);
```

Comment:
Please add tests that exercise this functionality.

Reply:
We already have that: line 62 in Strings_UnitTest.cpp

Reply:
I believe that that test doesn't expose the problem because it is using string iterators as the underlying type; we need to have another test that uses a different iterator type (char pointers for example).

Reply:
have undo this